src/net/tls.cc

```diff
@@ -38,14 +38,21 @@
 #include <seastar/net/stack.hh>
 #include <seastar/util/std-compat.hh>
 #include <seastar/util/variant_utils.hh>
+#include <seastar/util/log.hh>
 
 #include <boost/range/iterator_range.hpp>
 #include <boost/range/adaptor/map.hpp>
 
 #include "../core/fsnotify.hh"
 
+#ifdef SEASTAR_DEBUG
+bool tls_inject_handshake_error = false;
+#endif
```

Comment:
I tried something like that first, but to reproduce the problem I had to to set the error only when `_out.put()` returned a non-ready future.  I'm not sure if that's doable with error-injecting pipe implementation.
I suggest to get the fix first and then wok on the test as a follow up since without it scylla tends to crash quite frequently in upgrade tests (https://github.com/scylladb/scylladb/issues/11252)